controllers/hetznercluster_controller_test.go

```diff
@@ -443,6 +443,44 @@ var _ = Describe("Hetzner ClusterReconciler", func() {
 					return isPresentAndFalseWithReason(key, instance, infrav1.LoadBalancerReadyCondition, infrav1.LoadBalancerFailedToOwnReason)
 				}, timeout, time.Second).Should(BeTrue())
 			})
+
+			It("should work with capi.syself.com/allow-empty-control-plane-address annotation error condition", func() {
+				instance.Annotations = make(map[string]string)
+				instance.Annotations[infrav1.AllowEmptyControlPlaneAddressAnnotation] = "true"
+				instance.Spec.ControlPlaneLoadBalancer.Enabled = false
+				instance.Spec.ControlPlaneEndpoint = &clusterv1.APIEndpoint{
```

Comment:
I believe that this should be nil right? :D

Reply:
yes and no it should work in both cases because the Kamaji operator uses the port to configure the loadbalancer.

Reply:
added both tests

Reply:
I'm sure that it should work, but this current setup works even without the flag you introduced.. What you want to test is that both Loadbalancer and ControlPlaneEndpoint are not set. Then it should only work with the flag (and the condition should be set on false accordingly).
I believe currently, your condition would be also on true, no? Maybe I'm missing something but that's what I understand from https://github.com/syself/cluster-api-provider-hetzner/pull/1106/files#diff-6ec10baeb9341142d245628229b63ac327258d210a8cf290fb19771006046aafR236

Reply:
ok I refactored the code a bit because to me it looked that there are some duplicated code snippets. There is one tests case which I can't create because the controller and the fake client are maybe to fast.

Loadbalancer enabled but no loadbalancer ready yet should create a false condition

Reply:
nevermind found a way

Reply:
oh that's a good catch! there was indeed duplicate code! thanks a lot for finding this!